### PR TITLE
indroduce Typescript

### PR DESCRIPTION
**Pull request by Simon-Laux**
Improved version of #1333, even if it is a bit bigger it works better now.
It doesn't include #1271 though so we need to look at resurrecting that pr after merging this one.

Also please review commit by commit - its easier

_Note: Credit for pushing this feature forward and doing the start goes to @nicodh_

### CODE REVIEW

File: src/renderer/components/map/MapComponent.js [hunk 1]
```diff
@@ -9,10 +9,9 @@ const geojsonExtent = require('@mapbox/geojson-extent')
 const moment = require('moment/moment')
 const formatRelativeTime = require('../conversations/formatRelativeTime')
 const MapLayerFactory = require('./MapLayerFactory')
-const { Slider, Button, Collapse } = require('@blueprintjs/core/lib/esm/index')
```
Comment (nicodh) [inline src/renderer/components/map/MapComponent.js]: Why did you remove this line?

Comment (nicodh) [reply src/renderer/components/map/MapComponent.js]: I readded it assuming it was by error removed

**Comment by nicodh** [comment]
I reviewed this commit by commit. Looks good to me and can be merged if it is rebased!

File: src/renderer/components/map/MapComponent.js (hunk 1 above)
Comment (Simon-Laux) [reply src/renderer/components/map/MapComponent.js]: I believe I changed it to the commonjs counterpart in the parcel pr and it slipped in that way whiles moving stuff back into the ts pr.
Good catch. That"s what reviewing is for :+1:
